docs: "What's new in Svelte" July newsletter (#2068)
sveltejs/svelte.dev: apps/svelte.dev/content/blog/2026-07-01-whats-new-in-svelte-july-2026.md

---
title: "What's new in Svelte: July 2026"
description: 'SvelteKit config in vite.config, explicit env vars and new declaration tag support across the toolchain'
author: Dani Sandoval
authorURL: https://dreamindani.com
---

This month brought a real shift in how SvelteKit projects are configured. You can now define your SvelteKit config directly inside `vite.config.js` and skip `svelte.config.js` entirely. We also got the first preview of explicit environment variables, which will eventually replace `$env/*` modules in SvelteKit 3.

On top of that, the language tools and the `sv` CLI both caught up with Svelte's new `{const ...}` declaration tags, so the whole toolchain is now in sync.

Let's dive in!

## What's new in SvelteKit

> - You can now pass your SvelteKit config directly to the Vite plugin, so a separate `svelte.config.js` is no longer required (**2.62.0**, [Docs](https://svelte.dev/docs/kit/configuration), [#15944](https://github.com/sveltejs/kit/pull/15944))

Reviewer: Removing svelte.config.js will also be required in Kit 3. Might need to mention this similar to how explicit environment variables was explained

Author: 👍 added a note in b4f2a81a8 that this is a preview of how Kit 3 will require config to live in `vite.config.js`.

- The Svelte CLI demo template now uses the new `{const ...}` declaration tag, so newly created projects show off the latest Svelte syntax (**sv@0.16.0**, [#1110](https://github.com/sveltejs/cli/pull/1110))
- `sv create` now scaffolds projects against `@sveltejs/kit` `^2.62.0` and moves the Svelte config into the Vite plugin by default (**sv@0.16.0**, [#1119](https://github.com/sveltejs/cli/pull/1119))
- A new experimental add-on lets you toggle experimental flags and opt into `@next` versions directly from the CLI (**sv@0.16.0**, [#1121](https://github.com/sveltejs/cli/pull/1121))
- The `drizzle` and `better-auth` add-ons now support SvelteKit's new explicit environment variables (**sv@0.16.0**, [#1122](https://github.com/sveltejs/cli/pull/1122))
- New `defineEnv` and `svelteConfig` helpers in `@sveltejs/sv-utils` make it easier to read and edit a project's Svelte config from add-ons (**sv-utils@0.3.0**)
- The Svelte language server, `svelte-check` and `svelte2tsx` now understand Svelte 5's `{const ...}` declaration tags (**svelte-language-server@0.18.1/svelte-check@4.5.0/svelte2tsx@0.7.56**, [#3033](https://github.com/sveltejs/language-tools/pull/3033))

> - The Svelte config can now be written in TypeScript with `svelte.config.ts` or `svelte.config.mts` (**svelte-language-server@0.18.1**, [#3009](https://github.com/sveltejs/language-tools/pull/3009))

Reviewer: Might be worth removing this point as the PR was merged in May and may not make sense when we also explain that the svelte config file is no longer needed earlier in the blog post

- `svelte-check` now accepts a `--config` option to point at a custom config file location (**svelte-check@4.7.0**, [#3066](https://github.com/sveltejs/language-tools/pull/3066))
- Experimental `tsgo` (TypeScript Go) support is now available in `svelte-check` for faster type checking on large codebases (**svelte-check@4.7.0**, [#3036](https://github.com/sveltejs/language-tools/pull/3036))

Want to dive deeper? Check out the [Svelte CLI](https://github.com/sveltejs/cli/releases) and [language-tools](https://github.com/sveltejs/language-tools/releases) releases. For all the minor changes and bugfixes that came out in the Svelte compiler this month, you can read the full [Svelte CHANGELOG](https://github.com/sveltejs/svelte/blob/main/packages/svelte/CHANGELOG.md).

---

## Community Showcase

### Apps & Sites built with Svelte

- [COLOR LAB](https://colorlab.ferreyrapons.com) is a browser-based color science instrument that explores RGB gamuts as 3D solids and builds perceptual theme ramps with WCAG checks

> - [Cometline](https://cometline.github.io) is a local desktop AI companion built with SvelteKit, Electron and a Go agent core ([GitHub](https://github.com/Cometline/cometline))

Reviewer: ```suggestion
- [Cometline](https://cometline.github.io) is a local desktop AI companion built with SvelteKit, Electron and a Go agent core
```
There's a "View on GitHub button" in the hero of the website so we may not need the explicit link

### Learning Resources

_Featuring Svelte Contributors and Ambassadors_

- [Agentic Engineering with Svelte](https://www.youtube.com/playlist?list=PLdAi-nB9AYut1ixrraX3E96rNNVj3iq4F) by Paolo Ricciuti is a new YouTube series from Mainmatter that documents building a real production Svelte app with AI agents (8 episodes so far, with an AMA on [July 15th](https://luma.com/2nqh8mvb))
- [My Favorite Framework Just Got Better](https://www.youtube.com/watch?v=j9fh4WC0zHo) by Joy of Code walks through realtime data in SvelteKit using remote functions, live queries and generators

_This Week in Svelte_

- [Ep. 143](https://www.youtube.com/watch?v=HCpuzfRdVvI) - Changelog, Mochi
- [Ep. 144](https://www.youtube.com/watch?v=uwA1d6neWQE) - Changelog

_To Read_

- [Picking Svelte in 2026: the honest tradeoff nobody tells you](https://dimonb19a.hashnode.dev/picking-svelte-in-2026) by Dmitrii Bormotov
- [Wuchale: One Year of Compile-Time i18n](https://dev.to/k1dv5/wuchale-one-year-of-compile-time-i18n-178a) by Kidus Adugna

### Libraries, Tools & Components

_Frameworks and Tooling_

> - [Mochi](https://mochi.fast) is a performance-focused alternative to SvelteKit from Stanislav Khromov that uses islands architecture and programmatic routing on Bun ([GitHub](https://github.com/khromov/mochi))

Reviewer: ```suggestion
- [Mochi](https://mochi.fast) is a performance-focused alternative to SvelteKit from Stanislav Khromov that uses islands architecture and programmatic routing on Bun
```
The site already has a GitHub button in the top-left of the nav bar so we could remove the explicit link

- [pottz](https://github.com/lucaletizia/pottz) bundles your SvelteKit app, the Bun runtime and a webview into a single executable so you can ship it as a native desktop app

> - [Svelte TV](https://github.com/rasterzero-dev/svelte-tv) renders Svelte components to WebGL instead of the DOM for use on smart TVs and low-memory webviews

Reviewer: ```suggestion
- [Svelte TV](https://rasterzero-dev.github.io/svelte-tv-docs/) renders Svelte components to WebGL instead of the DOM for use on smart TVs and low-memory webviews
```
There's an official website for this one

- [TabSpot](https://github.com/JLAcostaEC/tabspot) is a declarative keyboard navigation engine for hierarchical and grid interfaces

_UI Components and Visual Effects_

> - [neobrutalism-svelte](https://neobrutalism-svelte.flenze.com) is a UI library inspired by neobrutalism, balsamiq and lo-fi aesthetics ([GitHub](https://github.com/olegpolin/neobrutalism-svelte))

Reviewer: ```suggestion
- [neobrutalism-svelte](https://neobrutalism-svelte.flenze.com) is a UI library inspired by neobrutalism, balsamiq and lo-fi aesthetics
```
The GitHub link is in the nav bar so we might not need it here

> - [@winkintel/bootstrap-svelte](https://bootstrap-svelte.vercel.app) brings Bootstrap 5's grid, utilities and components to Svelte 5 with full TypeScript support ([GitHub](https://github.com/WinkIntel/bootstrap-svelte))

Reviewer: ```suggestion
- [@winkintel/bootstrap-svelte](https://bootstrap-svelte.vercel.app) brings Bootstrap 5's grid, utilities and components to Svelte 5 with full TypeScript support
```
Same here

- [Tan Compose](https://ra9.github.io/tan-compose) is a lightweight library for building declarative reusable web components
- [Svaul](https://harshmandan.github.io/svaul/) is a zero-dependency drawer component for Svelte 5, modernizing vaul-svelte
- [Svelte Animated Icon](https://svelte-animated-icon.vercel.app) ships close to 10,000 animated icon variants across five icon libraries using the Web Animations API ([GitHub](https://github.com/fractalmandala/svelte-animated-icon))
- [FlareCharts](https://faintshadow.gitlab.io/flarecharts) is a runes-native charting library for Svelte 5 with built-in CSS theming and accessibility

> - [Svelte Video Editor](https://svelte-video-editor.ariefsn.dev) is a host-agnostic timeline-based video editor component for Svelte 5 with multi-track clips, scrubbing and ripple edits ([GitHub](https://github.com/ariefsn/svelte-video-editor))

Reviewer: ```suggestion
- [Svelte Video Editor](https://svelte-video-editor.ariefsn.dev) is a host-agnostic timeline-based video editor component for Svelte 5 with multi-track clips, scrubbing and ripple edits
```

_Developer Tools and Plugins_

> - [tsv](https://tsv.fuz.dev) is a Rust-based formatter and parser for TypeScript, Svelte and CSS, with a linter on the roadmap ([GitHub](https://github.com/fuzdev/tsv))

Reviewer: ```suggestion
- [tsv](https://tsv.fuz.dev) is a Rust-based formatter and parser for TypeScript, Svelte and CSS, with a linter on the roadmap
```

> - [svelte-docinfo](https://svelte-docinfo.fuz.dev) extracts JSON documentation from TypeScript and Svelte modules using the TypeScript compiler ([GitHub](https://github.com/fuzdev/svelte-docinfo))

Reviewer: ```suggestion
- [svelte-docinfo](https://svelte-docinfo.fuz.dev) extracts JSON documentation from TypeScript and Svelte modules using the TypeScript compiler
```
